src/sentry/integrations/discord/integration.py

```diff
@@ -238,39 +237,19 @@ def _get_discord_user_id(self, auth_code: str, url: str) -> str:
         integration.
 
         """
-        form_data = f"client_id={self.application_id}&client_secret={self.client_secret}&grant_type=authorization_code&code={auth_code}&redirect_uri={url}"
-        headers = {
-            "Content-Type": "application/x-www-form-urlencoded",
-        }
-
         try:
-            response = self.client.post(
-                "https://discord.com/api/v10/oauth2/token",
-                json=False,
-                data=form_data,
-                headers=headers,
-            )
-            token = response["access_token"]  # type: ignore
-
+            access_token = self.client.get_access_token(auth_code, url)
+            user_id = self.client.get_user_id(access_token)
+            return user_id
```

Comment:
@ceorourke Should I try to separate these to differentiate between access token errors vs. user id errors?

Reply:
Yeah, I think that's a good idea.

Reply:
@ceorourke Do the changes I just made properly handle the errors?